Realm/RLMRealmConfiguration.mm

```diff
@@ -236,5 +236,16 @@ - (void)setCustomSchema:(RLMSchema *)customSchema {
     _config.schema = [_customSchema objectStoreCopy];
 }
 
+- (void)setDisableFormatUpgrade:(bool)disableFormatUpgrade
+{
+    _config.disable_format_upgrade = disableFormatUpgrade;
+    _config.cache = false;
```

Comment:
In `-setDynamic:` we only disable caching if we're setting the value to the non-default value. I think we be consistent between these two methods.